src/stan/model/indexing/assign_varmat.hpp

```diff
@@ -84,34 +79,31 @@ inline void assign(VarVec&& x,
  * @tparam Vec2 `var_value` with inner Eigen type with either dynamic rows or
  * columns, but not both.
  * @param[in] x Vector to be assigned.
- * @param[in] idxs Index holding an `std::vector` of cells to assign to.
  * @param[in] y Value vector.
- * @param[in] name Name of variable (default "ANON").
- * @param[in] depth Indexing depth (default 0).
+ * @param[in] name Name of variable
+ * @param[in] idx Index holding an `std::vector` of cells to assign to.
  * @throw std::out_of_range If any of the indices are out of bounds.
  * @throw std::invalid_argument If the value size isn't the same as
  * the indexed size.
  */
 template <typename Vec1, typename Vec2,
           require_all_var_vector_t<Vec1, Vec2>* = nullptr>
-inline void assign(Vec1&& x,
-                   const cons_index_list<index_multi, nil_index_list>& idxs,
-                   const Vec2& y, const char* name = "ANON", int depth = 0) {
+inline void assign(Vec1&& x, const Vec2& y, const char* name,
+                   const index_multi& idx) {
   stan::math::check_size_match("vector[multi] assign", "left hand side",
-                               idxs.head_.ns_.size(), name, y.size());
+                               idx.ns_.size(), name, y.size());
   const auto x_size = x.size();
-  const auto assign_size = idxs.head_.ns_.size();
+  const auto assign_size = idx.ns_.size();
   arena_t<std::vector<int>> x_idx(assign_size);
   arena_t<Eigen::Matrix<double, -1, 1>> prev_vals(assign_size);
   Eigen::Matrix<double, -1, 1> y_vals(assign_size);
   std::unordered_set<int> x_set;
   x_set.reserve(assign_size);
   // We have to use two loops to avoid aliasing issues.
   for (int i = assign_size - 1; i >= 0; --i) {
-    if (likely(x_set.insert(idxs.head_.ns_[i]).second)) {
-      stan::math::check_range("vector[multi] assign", name, x_size,
-                              idxs.head_.ns_[i]);
-      x_idx[i] = idxs.head_.ns_[i] - 1;
+    if (likely(x_set.insert(idx.ns_[i]).second)) {
```

Comment:
I wonder if we should remove the likely here? I've been writing things lately that have a lot of repeated values so idk it's a judgement call

Reply:
No idea.